File: LoanCalculator/LoanAction.swift
//
//  LoanAction.swift
//  LoanCalculator
//
//  Created by Ivan Tishchenko on 17.12.2025.
//

import Foundation

// MARK: - Loan Actions
enum LoanAction {
    case setAmount(Double)
    case setPeriod(Int)
    case submitApplication
    case submissionStarted
    case submissionSuccess(responseId: Int)
    case submissionFailure(message: String)
    case dismissResult
    case loadSavedState
}

File: Android/app/src/main/java/com/loancalculator/redux/LoanAction.kt
package com.loancalculator.redux

/**
 * Loan Actions - all possible actions in the loan calculator
 */
sealed class LoanAction {
    data class SetAmount(val amount: Double) : LoanAction()
    data class SetPeriod(val period: Int) : LoanAction()
    object SubmitApplication : LoanAction()
    object SubmissionStarted : LoanAction()
    data class SubmissionSuccess(val responseId: Int) : LoanAction()
    data class SubmissionFailure(val message: String) : LoanAction()
    object DismissResult : LoanAction()
    object LoadSavedState : LoanAction()
}

